test/controllers/CheckDetailsControllerSpec.scala

```diff
@@ -367,97 +411,148 @@ class CheckDetailsControllerSpec extends SpecBase with SummaryListFluency {
+            .thenReturn(Future.successful(mockPDVResponseData))
+          when(mockIndividualDetailsConnector.getIndividualDetails(any(), anyValueType[ResolveMerge])(any(), any(), anyValueType[CorrelationId]))
+            .thenReturn(IndividualDetailsResponseEnvelope(Right(fakeIndividualDetailsWithMatchingPostcode)))
 
-  val fakeAddress: Address = Address(
-    addressSequenceNumber = AddressSequenceNumber(0),
-    addressSource = Some(AddressSource.Customer),
-    countryCode = CountryCode(826), // 826 is the numeric code for the United Kingdom
-    addressType = AddressType.ResidentialAddress,
-    addressStatus = Some(AddressStatus.NotDlo),
-    addressStartDate = LocalDate.of(2000, 1, 1),
-    addressEndDate = Some(LocalDate.of(2022, 12, 31)),
-    addressLastConfirmedDate = Some(LocalDate.of(2022, 1, 1)),
-    vpaMail = Some(VpaMail(1)),
-    deliveryInfo = Some(DeliveryInfo("Delivery info")),
-    pafReference = Some(PafReference("PAF reference")),
-    addressLine1 = AddressLine("123 Fake Street"),
-    addressLine2 = AddressLine("Apt 4B"),
-    addressLine3 = Some(AddressLine("Faketown")),
-    addressLine4 = Some(AddressLine("Fakeshire")),
-    addressLine5 = Some(AddressLine("Fakecountry")),
-    addressPostcode = Some(AddressPostcode("AA1 1AA"))
-  )
+          val app = applicationBuilder(userAnswers = Some(emptyUserAnswers))
+            .overrides(
+              inject.bind[IndividualDetailsConnector].toInstance(mockIndividualDetailsConnector),
+              inject.bind[PersonalDetailsValidationService].toInstance(mockPersonalDetailsValidationService)
+            )
+            .build()
 
-  val fakeIndividualDetails: IndividualDetails = IndividualDetails(
-    ninoWithoutSuffix = "AB123456",
-    ninoSuffix = Some(NinoSuffix("C")),
-    accountStatusType = Some(AccountStatusType.FullLive),
-    dateOfEntry = Some(LocalDate.of(2000, 1, 1)),
-    dateOfBirth = LocalDate.of(1990, 1, 1),
-    dateOfBirthStatus = Some(DateOfBirthStatus.Verified),
-    dateOfDeath = None,
-    dateOfDeathStatus = None,
-    dateOfRegistration = Some(LocalDate.of(2000, 1, 1)),
-    crnIndicator = CrnIndicator.False,
-    nameList = NameList(Some(List(fakeName))),
-    addressList = AddressList(Some(List(fakeAddress)))
-  )
+          val controller = app.injector.instanceOf[CheckDetailsController]
 
-  val mockAuthConnector: AuthConnector = mock[AuthConnector]
-  val mockIndividualDetailsConnector: IndividualDetailsConnector = mock[IndividualDetailsConnector]
-  val mockPersonalDetailsValidationService: PersonalDetailsValidationService = mock[PersonalDetailsValidationService]
+          running(app) {
+            val request = FakeRequest(GET, routes.CheckDetailsController.onPageLoad(ivOrigin, NormalMode).url)
+              .withSession("sessionId" -> "", "credentialId" -> "")
+
+            val resp = controller.onPageLoad(ivOrigin, NormalMode)(request)
+            status(resp) mustEqual 303
```

Comment:
As above. This test does not assert the redirect (InvalidDataNINOHelpController) as mentioned in the test name.